Sources/Datadog/RUMMonitor.swift

```diff
@@ -607,6 +608,25 @@ public class RUMMonitor: DDRUMMonitor, RUMCommandSubscriber {
                 }
             }
         }
+
+        // update the core context with rum context
+        core.set(feature: "rum", attributes: self.queue.sync {
```

Comment:
Because `process(command:)` is called from the user thread, `queue.sync {}` here will be blocking it. This is due to race conditions we have discovered in SDK context and that we have an open RFC on, right 🙂?

Reply:
The `attributes` from `core.set(feature:.attributes:)` is an `@escaping @autoclosure` that is called from the context queue. This syntax is very confusing I admit... 2 options:
1. explicit `@autoclosure`:
```swift
core.set(feature: "rum", attributes: { 
   return self.queue.sync {
        ...
    }
}())
```
2. Remove `@autoclosure`, making `@escaping` explicit.
```swift
core.set(feature: "rum", attributes: { 
    return self.queue.sync {
        ...
    }
})
```
Closure will have always be explicit.

That second will read better, WDYT?

Reply:
Oh, definitely it was too misleading, so I was trapped. Yes, `@autoclosure` removal we did looks good 👍.